Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -181,4 +181,31 @@ extension File {
         }
         return violatingRanges
     }
+
+    internal func numberOfCommentOnlyLines(startLine: Int, endLine: Int) -> Int {
```

Comment:
They do, but probably they name could be changed to reflect that.

`syntaxKindsByLine()` adds the whitespace only lines with an empty `SyntaxKind` array. Then `numberOfCommentOnlyLines` keeps these lines because of
```swift
kinds.filter { !commentKinds.contains($0) }.isEmpty
```
as `kinds` will be empty, `.filter` won't have any effect, keeping the line on the resulting array of the outer `.filter`.